src/cargo/ops/registry.rs

```diff
@@ -442,13 +443,26 @@ fn wait_for_publish(
 ) -> CargoResult<()> {
     let version_req = format!("={}", pkg.version());
     let mut source = SourceConfigMap::empty(config)?.load(registry_src, &HashSet::new())?;
-    let source_description = source.describe();
+    // Disable the source's built-in progress bars. Repeatedly showing a bunch
+    // of independent progress bars can be a little confusing. There is an
+    // overall progress bar managed here.
+    source.set_quiet(true);
+    let source_description = source.source_id().to_string();
     let query = Dependency::parse(pkg.name(), Some(&version_req), registry_src)?;
 
     let now = std::time::Instant::now();
     let sleep_time = std::time::Duration::from_secs(1);
-    let mut logged = false;
-    loop {
+    let max = timeout.as_secs() as usize;
+    config.shell().status("Published", pkg.to_string())?;
```

Comment:
I was thinking about this PR when publishing a crate just now and the thought that hit me is that this isn't quite accurate.  It isn't published yet; it is enqueued for publish.

Reply:
Hm, how about changing things from:

```
Uploading foo v1.0.0
Published foo v1.0.0
...
Completed foo v1.0.0 ...
```

to:

```
Uploading foo v1.0.0
Uploaded foo v1.0.0 has been successfully uploaded to registry `crates-io`
...
Published foo v1.0.0 has been successfully published to registry `crates-io`
```

Reply:
I think `Uploaded` and `Published` work.  I can also see discussing it as queuing.

However, that sample text reads oddly.  Shouldn't it be
```
Uploading foo v1.0.0
Uploaded foo v1.0.0 to registry `crates-io`
...
Published foo v1.0.0 at registry `crates-io`
```
- "successfully ..." was reptitive with the leading word
- "Published to" makes it sound like we were still actively involved.  My hope is "Published at" lessons that.

Reply:
Sounds good! I pushed an updated with that.